GoMeet/src/main/java/com/google/sps/servlets/MeetingEventServlet.java

```diff
@@ -0,0 +1,133 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.sps.data.ErrorMessages;
+import com.google.sps.data.MeetingEventFields;
+import com.google.sps.data.ServletUtil;
+import org.json.simple.JSONObject;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.HashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet to handle the fetching and creating of meetingEvent entities */
+@WebServlet("/meeting-event")
+public class MeetingEventServlet extends HttpServlet {
+
+  /** Creates a new meetingEvent entity and stores it to Datastore */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); 
+    String meetingName = request.getParameter(MeetingEventFields.MEETING_NAME);
+    String durationMins = request.getParameter(MeetingEventFields.DURATION_MINS); 
+    String durationHours = request.getParameter(MeetingEventFields.DURATION_HOURS); 
+    String timeFindMethod = request.getParameter(MeetingEventFields.TIME_FIND_METHOD);
+    String guestListUri = request.getParameter(MeetingEventFields.GUEST_LIST); 
+    String meetingTimeIdsUri = request.getParameter(MeetingEventFields.MEETING_TIME_IDS); 
+    String meetingLocationIdsUri = request.getParameter(MeetingEventFields.MEETING_LOCATION_IDS); 
+
+    // Check all properties are not null 
+    if (meetingName == null || durationMins == null || durationHours == null || 
+        timeFindMethod == null || guestListUri == null || meetingTimeIdsUri == null || 
+        meetingLocationIdsUri == null) {
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_BAD_REQUEST, 
+          ErrorMessages.BAD_POST_REQUEST_ERROR);
+      return;
+    }
+
+    // guestListStr, meetingTimeIdsStr and meetingLocationIdsStr are sent by the request 
+    // as a single string with each element deliminated by a comma 
+    String guestListStr = ServletUtil.decodeUri(guestListUri);
+    String meetingTimeIdsStr = ServletUtil.decodeUri(meetingTimeIdsUri);
+    String meetingLocationIdsStr = ServletUtil.decodeUri(meetingLocationIdsUri);
+    List<String> guestList = Arrays.asList(guestListStr.split(",")); 
+    List<String> meetingTimeIds = Arrays.asList(meetingTimeIdsStr.split(",")); 
+    List<String> meetingLocationIds = Arrays.asList(meetingLocationIdsStr.split(","));
+
+    // Create entity and store in Datastore 
+    Entity meetingEvent = new Entity("MeetingEvent"); 
```

Comment:
Optional: Maybe you could use a private helper function to add an entity to datastore. Not sure if it will make THAT big of a difference but if you move the Datastore API calls to another function, if there are any changes to the way the database is structured, it will be easier to find what code to update. :)))

Reply:
Done! Thanks for the suggestion :))